src/OrchardCore.Modules/OrchardCore.Tenants/Services/TenantValidator.cs

```diff
@@ -84,20 +86,35 @@ public async Task<IEnumerable<ModelError>> ValidateAsync(TenantViewModel model)
 
             var allOtherShells = allSettings.Where(t => !String.Equals(t.Name, model.Name, StringComparison.OrdinalIgnoreCase));
 
-            if (allOtherShells.Any(tenant => String.Equals(tenant.RequestUrlPrefix, model.RequestUrlPrefix?.Trim(), StringComparison.OrdinalIgnoreCase) && IsUrlHostExist(tenant.RequestUrlHost, model.RequestUrlHost)))
+            if (allOtherShells.Any(tenant => String.Equals(tenant.RequestUrlPrefix, model.RequestUrlPrefix?.Trim(), StringComparison.OrdinalIgnoreCase) && DoesUrlHostExist(tenant.RequestUrlHost, model.RequestUrlHost)))
             {
                 errors.Add(new ModelError(nameof(model.RequestUrlPrefix), S["A tenant with the same host and prefix already exists."]));
             }
 
             return errors;
         }
 
-        private static bool IsUrlHostExist(string urlHost, string modelUrlHost)
+        private static bool DoesUrlHostExist(string urlHost, string modelUrlHost)
+        {
+            if (String.IsNullOrEmpty(urlHost) && String.IsNullOrEmpty(modelUrlHost))
+            {
+                return true;
+            }
```

Comment:
Why is this restriction necessary? Creating a tenant with an empty host is valid.

Reply:
It's necessary to avoid NRE while doing `Intersect`, the easiest way to return `true` to indicate that there is an already empty host

Reply:
Can you create a tenant with an empty host with this in place?

Reply:
Please test, both manually and automatically.